lib/spack/spack/cmd/python.py

```diff
@@ -48,10 +51,47 @@ def python(parser, args, unknown_args):
     if unknown_args:
         tty.die("Unknown arguments:", " ".join(unknown_args))
 
+    # If arguments or a command are provided, just use python
+    if args.python_args or args.python_command:
+        args.python_interpreter = "python"
+
+    # Run user choose of interpreter
+    if args.python_interpreter == "ipython":
+        return spack.cmd.python.ipython_interpreter(args)
+    return spack.cmd.python.python_interpreter(args)
+
+
+def ipython_interpreter(args):
+    """An ipython interpreter is intended to be interactive, so it doesn't
+    support running a script or arguments
+    """
+    # If the user doesn't have ipython, fallback to python
+    try:
+        import IPython
+    except ImportError:
+        tty.warning("ipython is not installed, using python.")
```

Comment:
I think this should just fail if `IPython` cannot be imported, so no `try`/`except` is needed here.

Reply:
How about we still catch the error, give the meaningful message, but then exit cleanly? If we just try to do the import and IPython isn't installed, the user will see the ugly traceback.